File: benchmarks/cpp/quadratic_pairs.cpp
#include <iostream>
#include <chrono>
#include <cstdlib>
long long f(long long n) {
    volatile long long c = 0;
    for (long long i = 0; i < n; i++)
        for (long long j = 0; j < n; j++)
            c += (i ^ j) & 1;
    return c;
}
int main(int argc, char** argv) {
    long long n = std::atoll(argv[1]);
    auto t0 = std::chrono::high_resolution_clock::now();
    volatile long long r = f(n);
    auto t1 = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double, std::milli> ms = t1 - t0;
    std::cout << "ELAPSED_MS=" << ms.count() << " RESULT=" << r << std::endl;
    return 0;
}

File: benchmarks/java/QuadraticPairs.java
public class QuadraticPairs {
    static long f(long n) {
        long c = 0;
        for (long i = 0; i < n; i++)
            for (long j = 0; j < n; j++)
                c += (i ^ j) & 1;
        return c;
    }
    public static void main(String[] args) {
        long n = Long.parseLong(args[0]);
        long t0 = System.nanoTime();
        long r = f(n);
        long t1 = System.nanoTime();
        System.out.println("ELAPSED_MS=" + (t1 - t0) / 1_000_000.0 + " RESULT=" + r);
    }
}
